src/Library/demos/Event Controllers/main.js

```diff
@@ -0,0 +1,103 @@
+import Gtk from "gi://Gtk";
+import Gdk from "gi://Gdk";
+import Gio from "gi://Gio";
+
+const window = workbench.window;
+const ctrl_button = workbench.builder.get_object("ctrl_button");
+const click_button = workbench.builder.get_object("click_button");
+const stack = workbench.builder.get_object("stack");
+const pic1 = workbench.builder.get_object("pic1");
+const pic2 = workbench.builder.get_object("pic2");
+const gesture_frame = workbench.builder.get_object("gesture_frame");
+const primary_button = workbench.builder.get_object("primary_button");
+const middle_button = workbench.builder.get_object("middle_button");
+const secondary_button = workbench.builder.get_object("secondary_button");
+
+const pic1_file = Gio.File.new_for_path(pkg.pkgdatadir).resolve_relative_path(
+  "Library/demos/Event Controllers/image1.png",
+);
+
+const pic2_file = Gio.File.new_for_path(pkg.pkgdatadir).resolve_relative_path(
+  "Library/demos/Event Controllers/image2.png",
+);
+
+pic1.file = pic1_file;
+pic2.file = pic2_file;
+
+let ctrl_pressed = false;
+
+// Key controller to detect when the Ctrl key is pressed and released
+const key_controller = new Gtk.EventControllerKey();
+window.add_controller(key_controller);
+key_controller.connect("key-pressed", (controller, keyval, keycode, state) => {
+  if (keyval === Gdk.KEY_Control_L || keyval === Gdk.KEY_Control_R) {
+    ctrl_pressed = true;
+  }
+});
+
+key_controller.connect("key-released", (controller, keyval, keycode, state) => {
+  if (keyval === Gdk.KEY_Control_L || keyval === Gdk.KEY_Control_R) {
+    ctrl_pressed = false;
+  }
+});
+
+// Gesture controller to detect when the button is clicked
+const gesture_primary_click = new Gtk.GestureClick();
+
+ctrl_button.connect("clicked", () => {
+  if (ctrl_pressed) {
+    ctrl_button.label = _("Ctrl+Click detected");
+    ctrl_button.add_css_class("suggested-action");
+  } else {
+    ctrl_button.label = _("Click detected");
+    ctrl_button.remove_css_class("suggested-action");
+  }
+});
```

Comment:
The event controller here is unused, and actually just demonstrating the `Gtk.Button::clicked` signal which uses an event controller internally.

Instead this event controller should be attached to something that usually doesn't respond to clicks, like a `Gtk.Box` with a `Gtk.Label` inside.